### PR TITLE
Improved ChibiOS support

### PR DESCRIPTION
**Pull request by podhrmic**
- [x] minor updates in the ChibiOS makefile
- [x] add support for Lia_1.1_chibios
- [x] add support for Lisa M chibios
- [ ] add support for Pixhawk chibios
- [x] add support for VMS ECU chibios (documentation on the wiki TBD)

### CODE REVIEW

**Comment by podhrmic** [comment]
Indeed not complete yet, but wanted to get the changes out already since I changed the makefile - does it work on Apogee @gautierhattenberger ?

**Comment by gautierhattenberger** [comment]
It is working for Apogee. I just renamed a variable with a more explicit name.

**Comment by gautierhattenberger** [comment]
@podhrmic Do you still have things to do for li[s]a and px4 boards ?

**Comment by podhrmic** [comment]
Yes, I am on travels so I haven't had a chance to test it yet:( Hopefully by end of the week.

**Comment by podhrmic** [comment]
FYI here is about the bootloader and ChibiOS: http://www.chibios.com/forum/viewtopic.php?f=3&t=417&start=10

**Comment by flixr** [comment]
It would be good to not duplicate the default options like leds, uarts, actuators... maybe we should put them in `<board>_defaults.makefile` files and include them in the locm3 and chibios versions?

**Comment by podhrmic** [comment]
I was thinking about that - but then I was like well, maybe the defaults are different for each flavor of the board? Lisa 2.0 vs Lisa 2.1 vs Lia? If we are reasonably sure that the defaults are the same then I would be happy merging them into one.

**Comment by podhrmic** [comment]
@gautierhattenberger can we mark this as "initial support" and merge as is? The PX4 compiles and blinks a LED, nothing else yet (have to properly configure the IO in board.h), Lia/Lisa seems to work fine.
The reason for the merge is that I won't get to finish and test the boards properly until early March (have to attend to other things now) and it would be nice to have it in master before some additional refactoring happens.

File: conf/Makefile.chibios [hunk 1]
```diff
@@ -39,7 +60,6 @@ Q=@
 RTOS_DEBUG ?= 0
 
 ifeq (,$(findstring $(RTOS_DEBUG),0 FALSE))
-$(info DEBUGGING ON)
```
Comment (gautierhattenberger) [inline conf/Makefile.chibios]: Why did you remove this info. I find it useful. Is it possible to at least keep the `DEBUGGING ON` ?

File: conf/flash_modes.xml [hunk 2]
```diff
@@ -47,6 +47,7 @@
       <board name="li[s]?a_mx_.*"/>
       <board name="px4fmu_.*"/>
       <board name="elle*"/>
+      <board name="lia_.*"/>
```
Comment (gautierhattenberger) [inline conf/flash_modes.xml]: lia board should be handled by the general lisa regexp above. If it is not the case, better fix it.

**Comment by gautierhattenberger** [comment]
If you can address the two comments, I'll merge this. FYI we are about to add the support for the STM32F7 based on ChibiOS for our new board. We wanted to fly with it a bit before, but the weather is terrible these days.

File: conf/Makefile.chibios (hunk 1 above)
Comment (podhrmic) [inline conf/Makefile.chibios]: I found it annoying - since most of the time I was compiling I had the debugging off. Maybe have it display the message only if the debugging is ON? And make it more descriptive - such as " ChibiOS debug mode is ON" - sounds good?

File: conf/flash_modes.xml (hunk 2 above)
Comment (podhrmic) [inline conf/flash_modes.xml]: that's what I thought, but frankly I didn't spend much time figuring it out - adding lia explicitly made the trick

**Comment by podhrmic** [comment]
@gautierhattenberger I ll clean it up a little more and also take care of @flixr 's comment, hopefully today. Excited for F7 based autopilot - do you have more info about it?

**Comment by gautierhattenberger** [comment]
See https://github.com/enacuavlab/paparazzi/tree/stm32f7 for the code. The board itself is a rather big one dedicated to scientific research and designed for easier integration.

File: conf/Makefile.chibios (hunk 1 above)
Comment (podhrmic) [inline conf/Makefile.chibios]: Done - prints a message only when the DEbugging is ON

File: conf/flash_modes.xml (hunk 2 above)
Comment (podhrmic) [inline conf/flash_modes.xml]: Confirmed - the regular expression `li[s]?a_[ms]?_.*` doesn't cover Lia boards. I don't think you can easily make a general enough regexpr, so I am happy with adding Lia board separately.

Comment (flixr) [inline conf/flash_modes.xml]: it should cover `lia_m_*` and `lia_s_*` boards... which one does it not cover?
But maybe it is easier adding them separately anyway...
